### PR TITLE
Extension based groovy support

### PR DESCRIPTION
**Pull request by vietj**

### CODE REVIEW

File: src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
+package io.vertx.lang.groovy;
```
Comment (jponge) [inline src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java]: Lacks a license header

File: src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class RetroCompatExtension {
+
+  // Make this private
```
Comment (jponge) [inline src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java]: So... 😉

File: src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  public static Object unwrap(Object obj) {
```
Comment (jponge) [inline src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java]: `@SuppressWarnings("unchecked")`

File: src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    } else if (obj instanceof List) {
+      return toJsonArray((List<Object>) obj);
+    } else if (obj instanceof GString) {
+      return obj.toString();
+    }
+    return obj;
+  }
+
+  // Make this private
+  private static Object toJsonElement(Object obj) {
```
Comment (jponge) [inline src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java]: `@SuppressWarnings("unchecked")`

File: src/main/java/io/vertx/lang/groovy/VertxTransformation.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
+package io.vertx.lang.groovy;
```
Comment (jponge) [inline src/main/java/io/vertx/lang/groovy/VertxTransformation.java]: Missing license header

**Comment by vietj** [comment]
do you mean the generated groovy code from examples or the links from the doc ?

File: src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java (hunk 1 above)
Comment (vietj) [inline src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java]: ok

File: src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java (hunk 2 above)
Comment (vietj) [inline src/main/java/io/vertx/lang/groovy/RetroCompatExtension.java]: actually it's an outdated comment :-)

File: src/main/java/io/vertx/lang/groovy/VertxTransformation.java (hunk 5 above)
Comment (vietj) [inline src/main/java/io/vertx/lang/groovy/VertxTransformation.java]: ok

**Pull request merged**